agenta-backend/agenta_backend/models/db_models.py

```diff
@@ -1,303 +1,396 @@
+    image = relationship("ImageDB")
+    deployment_id = Column(Integer)  # reference to deployment, can be nullable
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "bases"
+
+
+class AppVariantDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    app_id = Column(Integer, ForeignKey("app_db.id"))
+    app = relationship("AppDB")
+    variant_name = Column(String)
+    revision = Column(Integer)
+    image_id = Column(Integer, ForeignKey("docker_images.id"))
+    image = relationship("ImageDB")
+    user_id = Column(Integer, ForeignKey("users.id"))
+    user = relationship("UserDB", foreign_keys=[user_id])
+    modified_by_id = Column(Integer, ForeignKey("users.id"))
+    modified_by = relationship("UserDB", foreign_keys=[modified_by_id])
+    base_name = Column(String)
+    base_id = Column(Integer, ForeignKey("bases.id"))
+    base = relationship("VariantBaseDB")
+    config_name = Column(String, nullable=False)
+    config_parameters = Column(JSON, nullable=False, default=dict)
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "app_variants"
+
+
+class AppVariantRevisionsDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    variant_id = Column(Integer, ForeignKey("app_variants.id"))
+    variant = relationship("AppVariantDB")
+    revision = Column(Integer)
+    modified_by_id = Column(Integer, ForeignKey("users.id"))
+    modified_by = relationship("UserDB")
+    base_id = Column(Integer, ForeignKey("bases.id"))
+    base = relationship("VariantBaseDB")
+    config_name = Column(String, nullable=False)
+    config_parameters = Column(JSON, nullable=False, default=dict)
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "app_variant_revisions"
+
+
+class AppEnvironmentDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    app_id = Column(Integer, ForeignKey("app_db.id"))
+    app = relationship("AppDB")
+    name = Column(String)
+    user_id = Column(Integer, ForeignKey("users.id"))
+    user = relationship("UserDB")
+    revision = Column(Integer)
+    deployed_app_variant_id = Column(Integer)  # reference to app_variant
+    deployed_app_variant_revision_id = Column(
+        Integer, ForeignKey("app_variant_revisions.id")
+    )
+    deployed_app_variant_revision = relationship("AppVariantRevisionsDB")
+    deployment_id = Column(Integer)  # reference to deployment
```

Comment:
same here, how are we using this in the code. Why not have a foreign_key and relationship if we have joins

Reply:
fixed